File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOWeatherSunTimeParamModel.swift
//
//  IDOWeatherSunTimeParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set sunrise and sunset time event number
@objcMembers
public class IDOWeatherSunTimeParamModel: NSObject, IDOBaseModel {
    /// Hour of sunrise   
    public var sunriseHour: Int
    /// Minute of sunrise 
    public var sunriseMin: Int
    /// Hour of sunset    
    public var sunsetHour: Int
    /// Minute of sunset  
    public var sunsetMin: Int
    
    enum CodingKeys: String, CodingKey {
        case sunriseHour = "sunrise_hour"
        case sunriseMin = "sunrise_min"
        case sunsetHour = "sunset_hour"
        case sunsetMin = "sunset_min"
    }
    
    public init(sunriseHour: Int,sunriseMin: Int,sunsetHour: Int,sunsetMin: Int) {
        self.sunriseHour = sunriseHour
        self.sunriseMin = sunriseMin
        self.sunsetHour = sunsetHour
        self.sunsetMin = sunsetMin
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOWeatherSunTimeParamModel.kt
//
//  IDOWeatherSunTimeParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Set sunrise and sunset time event number

open class IDOWeatherSunTimeParamModel(
    sunriseHour: Int,
    sunriseMin: Int,
    sunsetHour: Int,
    sunsetMin: Int
) : IDOBaseModel {

    /// Hour of sunrise   
    @SerializedName("sunrise_hour")
    var sunriseHour: Int = sunriseHour

    /// Minute of sunrise
    @SerializedName("sunrise_min")
    var sunriseMin: Int = sunriseMin

    /// Hour of sunset
    @SerializedName("sunset_hour")
    var sunsetHour: Int = sunsetHour

    /// Minute of sunset
    @SerializedName("sunset_min")
    var sunsetMin: Int = sunsetMin


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    